packages/core/saved-objects/core-saved-objects-migration-server-internal/src/initial_state.ts

```diff
@@ -117,18 +125,28 @@ export const createInitialState = ({
     );
   }
 
+  const targetIndexMappings: IndexMapping = {
+    ...targetMappings,
+    _meta: {
+      ...targetMappings._meta,
+      indexTypesMap,
+    },
+  };
+
   return {
     controlState: 'INIT',
     waitForMigrationCompletion,
+    mustRelocateDocuments,
+    indexTypesMap,
     indexPrefix,
     legacyIndex: indexPrefix,
     currentAlias: indexPrefix,
     versionAlias: `${indexPrefix}_${kibanaVersion}`,
     versionIndex: `${indexPrefix}_${kibanaVersion}_001`,
-    tempIndex: `${indexPrefix}_${kibanaVersion}_reindex_temp`,
+    tempIndex: getTempIndexName(indexPrefix, kibanaVersion),
     kibanaVersion,
     preMigrationScript: Option.fromNullable(preMigrationScript),
-    targetIndexMappings: targetMappings,
+    targetIndexMappings,
```

Comment:
Self review: **IMPORTANT** I must make sure that `migrationMappingPropertyHashes` only contains the types that belong to each migrator. Otherwise, changes in a type will cause all migrators to reindex

Reply:
Will these all still live on `.kibana`?

Reply:
I confirm that each of the indices' mappings only contains the ROOT_TYPES (core) + the types that belong to the index.